--- compiler/cpp/compiler.sln	Mon Jan 01 02:55:35 2018
+++ compiler/cpp/compiler.sln	Mon Oct 02 20:38:17 2017
@@ -2,2 +2,4 @@
-Microsoft Visual Studio Solution File, Format Version 11.00
-# Visual Studio 2010
+Microsoft Visual Studio Solution File, Format Version 12.00
+# Visual Studio 14
+VisualStudioVersion = 14.0.25420.1
+MinimumVisualStudioVersion = 10.0.40219.1
@@ -8,0 +11 @@
+		Debug|x64 = Debug|x64
@@ -9,0 +13 @@
+		Release|x64 = Release|x64
@@ -13,0 +18,2 @@
+		{89975A1A-F799-4556-98B8-64E30AB39A90}.Debug|x64.ActiveCfg = Debug|x64
+		{89975A1A-F799-4556-98B8-64E30AB39A90}.Debug|x64.Build.0 = Debug|x64
@@ -15,0 +22,2 @@
+		{89975A1A-F799-4556-98B8-64E30AB39A90}.Release|x64.ActiveCfg = Release|x64
+		{89975A1A-F799-4556-98B8-64E30AB39A90}.Release|x64.Build.0 = Release|x64
