diff --git a/bind-9.9.2-P1/bin/named/query.c b/bind-9.9.2-P1/bin/named/query.c
index 7ef0c9f..8f69ed9 100644
--- a/bind-9.9.2-P1/bin/named/query.c
+++ b/bind-9.9.2-P1/bin/named/query.c
@@ -733,8 +733,15 @@ query_validatezonedb(ns_client_t *client, dns_name_t *name,
 	 * zone content is not public data, but a part of local configuration
 	 * and should not be disclosed.
 	 */
+#ifdef CONTRAIL_DNS
+	dns_view_t *next_virtual_dns = dns_zone_getnextview(zone);
 	if (dns_zone_gettype(zone) == dns_zone_staticstub &&
+	    (!next_virtual_dns || !is_view_valid(next_virtual_dns)) &&
 	    !RECURSIONOK(client)) {
+#else
+	if (dns_zone_gettype(zone) == dns_zone_staticstub &&
+	    !RECURSIONOK(client)) {
+#endif
 		return (DNS_R_REFUSED);
 	}
 
