diff --git a/boost_1_48_0/boost/regex/v4/match_results.hpp b/boost_1_48_0/boost/regex/v4/match_results.hpp
index ca9898f..3bc9f42 100644
--- a/boost_1_48_0/boost/regex/v4/match_results.hpp
+++ b/boost_1_48_0/boost/regex/v4/match_results.hpp
@@ -556,7 +556,11 @@ private:
    static void raise_logic_error()
    {
       std::logic_error e("Attempt to access an uninitialzed boost::match_results<> class.");
+#ifndef BOOST_NO_EXCEPTIONS
       boost::throw_exception(e);
+#else
+    assert(false);
+#endif
    }
 
 
