diff --git a/boost_1_48_0/boost/filesystem/v3/operations.hpp b/boost_1_48_0/boost/filesystem/v3/operations.hpp
index c64f376..1adf5de 100644
--- a/boost_1_48_0/boost/filesystem/v3/operations.hpp
+++ b/boost_1_48_0/boost/filesystem/v3/operations.hpp
@@ -892,11 +892,15 @@ namespace detail
       const std::string & what_arg, system::error_code ec)
       : system::system_error(ec, what_arg)
     {
+#ifndef BOOST_NO_EXCEPTIONS
       try
+#endif
       {
         m_imp_ptr.reset(new m_imp);
       }
+#ifndef BOOST_NO_EXCEPTIONS
       catch (...) { m_imp_ptr.reset(); }
+#endif
     }
 
     filesystem_error(
@@ -904,12 +908,16 @@ namespace detail
       system::error_code ec)
       : system::system_error(ec, what_arg)
     {
+#ifndef BOOST_NO_EXCEPTIONS
       try
+#endif
       {
         m_imp_ptr.reset(new m_imp);
         m_imp_ptr->m_path1 = path1_arg;
       }
+#ifndef BOOST_NO_EXCEPTIONS
       catch (...) { m_imp_ptr.reset(); }
+#endif
     }
     
     filesystem_error(
@@ -917,13 +925,17 @@ namespace detail
       const path& path2_arg, system::error_code ec)
       : system::system_error(ec, what_arg)
     {
+#ifndef BOOST_NO_EXCEPTIONS
       try
+#endif
       {
         m_imp_ptr.reset(new m_imp);
         m_imp_ptr->m_path1 = path1_arg;
         m_imp_ptr->m_path2 = path2_arg;
       }
+#ifndef BOOST_NO_EXCEPTIONS
       catch (...) { m_imp_ptr.reset(); }
+#endif
     }
 
     ~filesystem_error() throw() {}
@@ -944,7 +956,9 @@ namespace detail
       if (!m_imp_ptr.get())
         return system::system_error::what();
 
+#ifndef BOOST_NO_EXCEPTIONS
       try
+#endif
       {
         if (m_imp_ptr->m_what.empty())
         {
@@ -964,10 +978,12 @@ namespace detail
         }
         return m_imp_ptr->m_what.c_str();
       }
+#ifndef BOOST_NO_EXCEPTIONS
       catch (...)
       {
         return system::system_error::what();
       }
+#endif
     }
 
   private:
