diff --git a/cpp-driver/CMakeLists.txt b/cpp-driver/CMakeLists.txt
index e459c3b..dacb69a 100644
--- a/cpp-driver/CMakeLists.txt
+++ b/cpp-driver/CMakeLists.txt
@@ -28,7 +28,7 @@ add_definitions(-DBOOST_ALL_DYN_LINK)
 add_definitions(-DBOOST_ALL_NO_LIB)
 
 set(Boost_USE_MULTITHREADED ON)
-find_package(Boost 1.41.0 COMPONENTS system thread unit_test_framework date_time program_options REQUIRED)
+find_package(Boost 1.41.0 COMPONENTS system thread date_time program_options REQUIRED)
 set(LIBS ${LIBS} ${Boost_LIBRARIES})
 
 find_package(OpenSSL REQUIRED)
@@ -200,15 +200,15 @@ add_custom_target(UNINSTALL
 #-------------------
 # Boost unit tests
 #-------------------
-add_subdirectory(test/unit_tests)
+# add_subdirectory(test/unit_tests)
 
 #-------------------
 # Integration tests
 #-------------------
-add_subdirectory(extra/ccm_bridge)
-add_subdirectory(test/integration_tests)
+# add_subdirectory(extra/ccm_bridge)
+# add_subdirectory(test/integration_tests)
 
 #-------------------
 # the demo program
 #-------------------
-add_subdirectory(demo)
+# add_subdirectory(demo)
diff --git a/cpp-driver/include/cql/cql_builder.hpp b/cpp-driver/include/cql/cql_builder.hpp
index 21a1e9e..64b0ea9 100644
--- a/cpp-driver/include/cql/cql_builder.hpp
+++ b/cpp-driver/include/cql/cql_builder.hpp
@@ -427,9 +427,6 @@ public:
     {
         boost::shared_ptr<boost::asio::ssl::context> ssl_context(
             new boost::asio::ssl::context(
-#if BOOST_VERSION <= 104800
-                _io_service,
-#endif
                 boost::asio::ssl::context::sslv23));
         _ssl_context = ssl_context;
         return *this;
