diff --git a/dpdk/lib/librte_eal/linuxapp/kni/ethtool/igb/igb_main.c b/dpdk/lib/librte_eal/linuxapp/kni/ethtool/igb/igb_main.c
index 0494780..b330b20 100644
--- a/dpdk/lib/librte_eal/linuxapp/kni/ethtool/igb/igb_main.c
+++ b/dpdk/lib/librte_eal/linuxapp/kni/ethtool/igb/igb_main.c
@@ -2276,7 +2276,12 @@ static int igb_ndo_bridge_getlink(struct sk_buff *skb, u32 pid, u32 seq,
 
 #ifdef HAVE_NDO_FDB_ADD_VID
 #ifdef HAVE_NDO_BRIDGE_GETLINK_NLFLAGS
+#ifdef HAVE_NDO_BRIDGE_GETLINK_FILTER_MASK_VLAN_FILL
+	return ndo_dflt_bridge_getlink(skb, pid, seq, dev, mode, 0, 0,
+				nlflags, filter_mask, NULL);
+#else
 	return ndo_dflt_bridge_getlink(skb, pid, seq, dev, mode, 0, 0, nlflags);
+#endif /* HAVE_NDO_BRIDGE_GETLINK_FILTER_MASK_VLAN_FILL */
 #else
 	return ndo_dflt_bridge_getlink(skb, pid, seq, dev, mode, 0, 0);
 #endif /* HAVE_NDO_BRIDGE_GETLINK_NLFLAGS */
diff --git a/dpdk/lib/librte_eal/linuxapp/kni/ethtool/igb/kcompat.h b/dpdk/lib/librte_eal/linuxapp/kni/ethtool/igb/kcompat.h
index e4fad18..5f45b8b 100644
--- a/dpdk/lib/librte_eal/linuxapp/kni/ethtool/igb/kcompat.h
+++ b/dpdk/lib/librte_eal/linuxapp/kni/ethtool/igb/kcompat.h
@@ -3901,4 +3901,9 @@ skb_set_hash(struct sk_buff *skb, __u32 hash, __always_unused int type)
 /* ndo_bridge_getlink adds new nlflags parameter */
 #define HAVE_NDO_BRIDGE_GETLINK_NLFLAGS
 #endif /* >= 4.1.0 */
+
+#if ( LINUX_VERSION_CODE >= KERNEL_VERSION(4,2,0) )
+/* ndo_bridge_getlink adds new filter_mask and vlan_fill parameters */
+#define HAVE_NDO_BRIDGE_GETLINK_FILTER_MASK_VLAN_FILL
+#endif /* >= 4.2.0 */
 #endif /* _KCOMPAT_H_ */
