--- ifmap-python-client/ifmap/client.py.original	2015-02-06 13:32:08.370889000 -0800
+++ ifmap-python-client/ifmap/client.py	2015-02-06 13:36:29.760376000 -0800
@@ -14,6 +14,7 @@
 
 import base64
 import cStringIO
+import _ssl
 
 
 
@@ -117,6 +118,10 @@
 		self.__url = url
                 self.__username = user
                 self.__password = password
+                self.__ssl_option_retried = False
+                self.http_conn()
+
+        def http_conn(self):
                 try:
                     self._http = HTTPClient(*self.__url, ssl = True,
                                         connection_timeout = None,
@@ -131,7 +136,6 @@
                                         ssl_options = self.__ssl_options,
                                         concurrency = concurrency)
 
-
 	def last_sent(self):
 		return self.__last_sent
 
@@ -171,7 +175,9 @@
 		  'SOAPAction': '%s' % (method),
 		}
 
-		try:
+                done = False
+                while not done:
+		    try:
 				log.info("sending IF-MAP message to server")
 				log.debug("========  sending IF-MAP message ========")
 				log.debug("\n%s\n", xml)
@@ -212,11 +218,25 @@
 
 				#return content
 				#pycurl return content.getvalue()
+                                done = True
 				return content
 
-		except	HttpException, e:
+		    except	HttpException, e:
 				log.error("HTTP Connection error in IF-MAP client: %s", e.reason)
-		except:
+				raise
+		    except	_ssl.SSLError:
+				log.error("SSL Connection error in IF-MAP client")
+                                if self.__ssl_option_retried:
+                                    done = True
+				    raise
+                                # retry once after toggling SSL version
+                                if self.__ssl_options['ssl_version'] == PROTOCOL_SSLv3:
+                                    self.__ssl_options['ssl_version'] = PROTOCOL_SSLv23             
+                                else:
+                                    self.__ssl_options['ssl_version'] = PROTOCOL_SSLv3             
+                                self.http_conn()
+                                self.__ssl_option_retried = True
+		    except:
 				log.error("Uknown error sending IF-MAP message to server")
 				raise
 
