diff --git a/third_party/ifmap-python-client/ifmap/client.py b/third_party/ifmap-python-client/ifmap/client.py
index c0f748c..103caa6 100644
--- a/third_party/ifmap-python-client/ifmap/client.py
+++ b/third_party/ifmap-python-client/ifmap/client.py
@@ -92,7 +92,14 @@
 		self.__url = url
                 self.__username = user
                 self.__password = password
-                self._http = HTTPClient(*self.__url, ssl = True,
+                try:
+                    self._http = HTTPClient(*self.__url, ssl = True,
+                                        connection_timeout = None,
+                                        network_timeout = None,
+                                        ssl_options = self.__ssl_options,
+                                        insecure = True)
+                except TypeError:
+                    self._http = HTTPClient(*self.__url, ssl = True,
                                         connection_timeout = None,
                                         network_timeout = None,
                                         ssl_options = self.__ssl_options)
