diff --git a/third_party/log4cplus-1.1.1/include/log4cplus/thread/syncprims.h b/third_party/log4cplus-1.1.1/include/log4cplus/thread/syncprims.h
index d043845..afe674e 100644
--- a/third_party/log4cplus-1.1.1/include/log4cplus/thread/syncprims.h
+++ b/third_party/log4cplus-1.1.1/third_party/include/log4cplus/thread/syncprims.h
@@ -291,6 +291,7 @@ void
 SyncGuard<SP>::attach_and_lock (SP const & m)
 {
     attach (m);
+#ifdef LOG4CPLUS_EXCEPTIONS
     try
     {
         lock();
@@ -300,6 +301,9 @@ SyncGuard<SP>::attach_and_lock (SP const & m)
         detach ();
         throw;
     }
+#else
+        lock();
+#endif
 }
 
 
