diff --git a/third_party/rabbitmq-c/librabbitmq/amqp_socket.c b/third_party/rabbitmq-c/librabbitmq/amqp_socket.c
index 66e4ae5..c4ae435 100644
--- a/third_party/rabbitmq-c/librabbitmq/amqp_socket.c
+++ b/third_party/rabbitmq-c/librabbitmq/amqp_socket.c
@@ -480,6 +480,12 @@ int amqp_open_socket_inner(char const *hostname,
       continue;
     }
 
+    if (0 != amqp_os_socket_setsockopt(sockfd, SOL_SOCKET, SO_KEEPALIVE, &one,
+                                     sizeof(one))) {
+      last_error = AMQP_STATUS_SOCKET_ERROR;
+      continue;
+    }
+
 #ifdef _WIN32
     res = connect(sockfd, addr->ai_addr, (int)addr->ai_addrlen);
 #else
