diff --git a/third_party/tbb40_20111130oss/src/tbb/mac64-tbb-export.lst b/third_party/tbb40_20111130oss/src/tbb/mac64-tbb-export.lst
--- a/third_party/tbb40_20111130oss/src/tbb/mac64-tbb-export.lst
+++ b/third_party/tbb40_20111130oss/src/tbb/mac64-tbb-export.lst
@@ -105,7 +105,7 @@ __TBB_SYMBOL( _ZTSN3tbb18captured_except
 __TBB_SYMBOL( _ZTVN3tbb18captured_exceptionE )
 __TBB_SYMBOL( _ZTIN3tbb13tbb_exceptionE )
 __TBB_SYMBOL( _ZTSN3tbb13tbb_exceptionE )
-__TBB_SYMBOL( _ZTVN3tbb13tbb_exceptionE )
+// __TBB_SYMBOL( _ZTVN3tbb13tbb_exceptionE ) macos linker issue
 #endif /* __TBB_TASK_GROUP_CONTEXT */
 
 // Symbols for exceptions thrown from TBB
--- /Users/ananth/work/bgp/third_party/tbb40_20111130oss/include/tbb/compat/condition_variable	2012-10-07 22:58:11.000000000 -0700
+++ include/tbb/compat/condition_variable	2013-12-10 14:20:22.000000000 -0800
diff --git a/third_party/tbb40_20111130oss/include/tbb/compat/condition_variable b/third_party/tbb40_20111130oss/include/tbb/compat/condition_variable
--- a/third_party/tbb40_20111130oss/include/tbb/compat/condition_variable
+++ b/third_party/tbb40_20111130oss/include/tbb/compat/condition_variable
@@ -435,6 +435,8 @@ __TBB_DEFINE_PROFILING_SET_NAME(interfac
 
 } // namespace tbb 
 
+#if defined(TBB_INJECT_INTO_STD_NAMESPACE)
+
 #if TBB_IMPLEMENT_CPP0X
 
 namespace std {
@@ -457,4 +459,6 @@ using tbb::interface5::no_timeout;
 
 #endif /* TBB_IMPLEMENT_CPP0X */
 
+#endif /* TBB_INJECT_INTO_STD_NAMESPACE */
+
 #endif /* __TBB_condition_variable_H */
